test/Swashbuckle.AspNetCore.IntegrationTests/SwaggerIntegrationTests.cs

```diff
@@ -112,5 +111,66 @@ public async Task SwaggerEndpoint_ReturnsCorrectPriceExample_ForDifferentCulture
             var json = await JsonSerializer.DeserializeAsync<JsonElement>(contentStream);
             Assert.Equal(expectedVersionValue, json.GetProperty(expectedVersionProperty).GetString());
         }
+
+#if NET8_0_OR_GREATER
+        [Theory]
+        [InlineData("/swagger/v1/swagger.json")]
+        public async Task SwaggerEndpoint_ReturnsValidSwaggerJson_For_WebApi(
+            string swaggerRequestUri)
+        {
+            await SwaggerEndpointReturnsValidSwaggerJson<Program>(swaggerRequestUri);
+        }
+
+        [Fact]
+        public async Task TypesAreRenderedCorrectly()
+        {
+            using var application = new TestApplication<Program>();
+            using var client = application.CreateDefaultClient();
+
+            using var swaggerResponse = await client.GetFromJsonAsync<JsonDocument>("/swagger/v1/swagger.json");
+
+            var weatherForecase = swaggerResponse.RootElement
+                .GetProperty("components")
+                .GetProperty("schemas")
+                .GetProperty("WeatherForecast");
+
+            Assert.Equal("object", weatherForecase.GetProperty("type").GetString());
```

Comment:
Nunit has support for doing [multiple assertions](https://docs.nunit.org/articles/nunit/writing-tests/assertions/multiple-asserts.html) in a test, might be worth wrapping all these in it so that if more than one regression occurs at once, they are identified in a single execution?

Reply:
We can maybe improve that in the future, but as we use xunit we wouldn't improve the assertions by pulling in NUnit too. Using something like Shouldly or FluentAssertions to extend xunit's capabilities would be better IMHO.

Reply:
Apologies, that was me being in a primarily nunit world, not thinking to check the library actually used here when checking the idea locally, but xunit also supports the same syntax.

Reply:
No problem - I'll take a look to see what there is. In my own projects I use Shouldly so I'm not used to using xunit's own built-in assertions.

Reply:
Thanks for the suggestion, I've refactored this to use `Assert.Multiple()`.